plugins/polio/migrations/0167_migrate_budget_data.py

```diff
@@ -0,0 +1,59 @@
+
+from django.db import migrations
+
+
+def migrate_data_forward(apps, schema_editor):
+    Campaign = apps.get_model("polio", "Campaign")
+    BudgetProcess = apps.get_model("polio", "BudgetProcess")
+
+    for campaign in Campaign.objects.all():
+        budget_steps = campaign.budget_steps.all()
+
+        if budget_steps.exists():
+            budget_process = BudgetProcess.objects.create(
+                created_at=budget_steps.first().created_at,
+                created_by=budget_steps.first().created_by,
+                updated_at=budget_steps.first().updated_at,
+                status=campaign.budget_status,
+                current_state_key=campaign.budget_current_state_key,
+                current_state_label=campaign.budget_current_state_label,
+                ra_completed_at_WFEDITABLE=campaign.ra_completed_at_WFEDITABLE,
+                who_sent_budget_at_WFEDITABLE=campaign.who_sent_budget_at_WFEDITABLE,
+                unicef_sent_budget_at_WFEDITABLE=campaign.unicef_sent_budget_at_WFEDITABLE,
+                gpei_consolidated_budgets_at_WFEDITABLE=campaign.gpei_consolidated_budgets_at_WFEDITABLE,
+                submitted_to_rrt_at_WFEDITABLE=campaign.submitted_to_rrt_at_WFEDITABLE,
+                feedback_sent_to_gpei_at_WFEDITABLE=campaign.feedback_sent_to_gpei_at_WFEDITABLE,
+                re_submitted_to_rrt_at_WFEDITABLE=campaign.re_submitted_to_rrt_at_WFEDITABLE,
+                submitted_to_orpg_operations1_at_WFEDITABLE=campaign.submitted_to_orpg_operations1_at_WFEDITABLE,
+                feedback_sent_to_rrt1_at_WFEDITABLE=campaign.feedback_sent_to_rrt1_at_WFEDITABLE,
+                re_submitted_to_orpg_operations1_at_WFEDITABLE=campaign.re_submitted_to_orpg_operations1_at_WFEDITABLE,
+                submitted_to_orpg_wider_at_WFEDITABLE=campaign.submitted_to_orpg_wider_at_WFEDITABLE,
+                submitted_to_orpg_operations2_at_WFEDITABLE=campaign.submitted_to_orpg_operations2_at_WFEDITABLE,
+                feedback_sent_to_rrt2_at_WFEDITABLE=campaign.feedback_sent_to_rrt2_at_WFEDITABLE,
+                re_submitted_to_orpg_operations2_at_WFEDITABLE=campaign.re_submitted_to_orpg_operations2_at_WFEDITABLE,
+                submitted_for_approval_at_WFEDITABLE=campaign.submitted_for_approval_at_WFEDITABLE,
+                feedback_sent_to_orpg_operations_unicef_at_WFEDITABLE=campaign.feedback_sent_to_orpg_operations_unicef_at_WFEDITABLE,
+                feedback_sent_to_orpg_operations_who_at_WFEDITABLE=campaign.feedback_sent_to_orpg_operations_who_at_WFEDITABLE,
+                approved_by_who_at_WFEDITABLE=campaign.approved_by_who_at_WFEDITABLE,
+                approved_by_unicef_at_WFEDITABLE=campaign.approved_by_unicef_at_WFEDITABLE,
+                approved_at_WFEDITABLE=campaign.approved_at_WFEDITABLE,
+                approval_confirmed_at_WFEDITABLE=campaign.approval_confirmed_at_WFEDITABLE,
+                payment_mode=campaign.payment_mode or "",
```

Comment:
why the or? Replacing None by blank spaces?

Reply:
Yes, because I used:

```python
payment_mode = models.CharField(max_length=30, choices=PAYMENT, blank=True)
```

The Django convention is to avoid using `null=True` for `CharField` and `TextField` since these fields could have two different values for “no data”: `None` or an empty string `""`.

https://sentry.io/answers/django-difference-between-null-and-blank/